runtime.md

```diff
@@ -29,24 +29,60 @@ The root directory to the bundle is provided in the state so that consumers can
 }
 ```
 
-## Lifecycle
+## Typical lifecycle
+
+A typical lifecyle progresses like this:
+
+1. There is no container
+2. A user tells the runtime to start a container and launch a process inside it
+3. The runtime [creates the container](#create)
+4. The runtime executes any [pre-start hooks](runtime-config.md#pre-start)
+5. The runtime [executes the container process](#start-process)
+6. The container process is running
+7. The runtime executes any [post-start hooks](runtime-config.md#post-start)
+8. A user tells the runtime to send a termination signal to the container process
+9. The runtime [sends a termination signal to the container process](#stop-process)
+10. The container process exits
+11. The runtime [terminates any other processes in the container](#stop-process)
+12. The runtime executes any [post-stop hooks](runtime-config.md#post-stop)
+13. The runtime [removes the container](#cleanup)
+
+With steps 7 and 8, the user is explicitly stopping the container process (via the runtime), but it's also possible that the container process could exit for other reasons.
+In that case we skip directly from 6 to [10](#stop-process), skipping any post-start hooks that hadn't been launched and terminating any in-progress post-start hook.
+
+Failure in a pre-start hook or other setup task can cause a jump straight to [12](runtime-config.md#post-stop).
 
 ### Create
 
-Creates the container: file system, namespaces, cgroups, capabilities.
+Create the container: file system, namespaces, cgroups, capabilities, etc.
+The invoked process forks, with one branch that stays in the host namespace and another that enters the container.
+The host process carries out all container setup actions, and continues running for the life of the container so it can perform teardown after the container process exits.
+The container process changes users and drops privileges in preparation for the container process start.
+At this point, the host process writes the [`state.json`](#state) file with the host-side version of the container-process's PID (the container process may be in a PID namespace).
 
 ### Start (process)
 
-Runs a process in a container.
-Can be invoked several times.
+After the pre-start hooks complete, the host process signals the container process to execute the runtime.
```

Comment:
This sentence does not make sense.  ` the host process signals the container process to execute the runtime.`????